ruamel.yaml new API implementation broke

<!-- Please fill in the following information to report a problem with Cantera. If you have a question about using Cantera, please post it on our Google Users' Group (https://groups.google.com/forum/#!forum/cantera-users). Feature enhancements should be discussed in the dedicated Cantera enhancements repository (https://github.com/Cantera/enhancements/new/choose) -->

**Problem description**

Some unit tests failed due to the breakage in the implementation of the new `ruamel.yaml` API. It may be related to #1048 and #1049. 

<!-- A clear and concise description of what the bug is. -->

**Steps to reproduce**

Run unit test on the main branch

```
======================================================================
FAIL: setUpClass (cantera.test.test_composite.TestModels)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\David\cantera\build\python\cantera\test\test_composite.py", line 17, in setUpClass
    cls.yml = utilities.load_yaml(cls.yml_file)
  File "C:\Users\David\cantera\build\python\cantera\test\utilities.py", line 25, in load_yaml
    return yaml_.load(stream)
  File "C:\Users\David\anaconda3\envs\cantera-dev\lib\site-packages\ruamel_yaml\main.py", line 331, in load
    return constructor.get_single_data()
  File "C:\Users\David\anaconda3\envs\cantera-dev\lib\site-packages\ruamel_yaml\constructor.py", line 111, in get_single_data
    return self.construct_document(node)
  File "C:\Users\David\anaconda3\envs\cantera-dev\lib\site-packages\ruamel_yaml\constructor.py", line 116, in construct_document
    data = self.construct_object(node)
  File "C:\Users\David\anaconda3\envs\cantera-dev\lib\site-packages\ruamel_yaml\constructor.py", line 167, in construct_object
    data = constructor(self, node)
  File "C:\Users\David\anaconda3\envs\cantera-dev\lib\site-packages\ruamel_yaml\constructor.py", line 729, in construct_undefined
    raise ConstructorError(
ruamel_yaml.constructor.ConstructorError: could not determine a constructor for the tag None
  in "C:\Users\David\cantera\build\python\cantera\test\data\thermo-models.yaml", line 1, column 1'
```

**System information**

- Cantera version:  2.6.0a2/ main branch
- OS: Windows 10
- Python/MATLAB/other software versions: Python 3.9

Got this error while trying to run the unit test. Seems the last two digits were switched.
```
======================================================================
FAIL: test_deprecated_callers (cantera.test.test_reaction.TestChebyshev)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\David\cantera\build\python\cantera\test\test_reaction.py", line 501, in test_deprecated_callers
    self.check_equal(rxn(T, P), value)
  File "C:\Users\David\cantera\build\python\cantera\test\test_reaction.py", line 458, in check_equal
    self.assertEqual(one, two)
AssertionError: 2858762454.1119056 != 2858762454.1119065
======================================================================
FAIL: test_deprecated_callers (cantera.test.test_reaction.TestChebyshev2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\David\cantera\build\python\cantera\test\test_reaction.py", line 498, in test_deprecated_callers
    self.check_equal(rxn(T, P), value)
  File "C:\Users\David\cantera\build\python\cantera\test\test_reaction.py", line 458, in check_equal
    self.assertEqual(one, two)
AssertionError: 2858762454.1119056 != 2858762454.1119065
----------------------------------------------------------------------
```

--- a/interfaces/cython/cantera/ck2yaml.py
+++ b/interfaces/cython/cantera/ck2yaml.py
@@ -1375,9 +1375,15 @@ class Parser:
         """
         Load YAML-formatted entries from ``path`` on disk.
         """
-        yaml_ = yaml.YAML()
-        with open(path, 'rt', encoding="utf-8") as stream:
-            yml = yaml_.load(stream)
+        try:
+            yaml_ = yaml.YAML(typ="rt")
+            with open(path, 'rt', encoding="utf-8") as stream:
+                yml = yaml_.load(stream)
+        except yaml.constructor.ConstructorError:
+            with open(path, "rt", encoding="utf-8") as stream:
+                # Ensure that the loader remains backward-compatible with legacy
+                # ruamel.yaml versions (prior to 0.17.0).
+                yml = yaml.round_trip_load(stream)
 
         # do not overwrite reserved field names
         reserved = {'generator', 'input-files', 'cantera-version', 'date',

--- a/interfaces/cython/cantera/test/test_reaction.py
+++ b/interfaces/cython/cantera/test/test_reaction.py
@@ -455,7 +455,7 @@ class ReactionTests:
         if isinstance(one, (list, tuple, np.ndarray)):
             self.assertArrayNear(one, two)
         else:
-            self.assertEqual(one, two)
+            self.assertNear(one, two)
 
     def test_deprecated_getters(self):
         # check property getters deprecated in new framework

--- a/interfaces/cython/cantera/test/utilities.py
+++ b/interfaces/cython/cantera/test/utilities.py
@@ -18,12 +18,17 @@ TEST_DATA_PATH = Path(__file__).parent / "data"
 CANTERA_DATA_PATH = Path(__file__).parents[1] / "data"
 
 
-def load_yaml(yml_file, typ="safe"):
-    # Load YAML data from file. The YAML loader defaults to "safe".
-    yaml_ = yaml.YAML(typ=typ)
-    with open(yml_file, "rt", encoding="utf-8") as stream:
-        return yaml_.load(stream)
-
+def load_yaml(yml_file):
+    # Load YAML data from file using the "safe" loading option.
+    try:
+        yaml_ = yaml.YAML(typ="safe")
+        with open(yml_file, "rt", encoding="utf-8") as stream:
+            return yaml_.load(stream)
+    except yaml.constructor.ConstructorError:
+        with open(yml_file, "rt", encoding="utf-8") as stream:
+            # Ensure that  the loader remains backward-compatible with legacy
+            # ruamel.yaml versions (prior to 0.17.0).
+            return yaml.safe_load(stream)
 
 class CanteraTest(unittest.TestCase):
     @classmethod

